### PR TITLE
Add remaining string.char_types APIs to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/strings/char_types.pyx [hunk 1]
```diff
@@ -1,4 +1,92 @@
...
     string_character_types as StringCharacterTypes  # no-cython-lint
+
+
+cpdef Column all_characters_of_type(
+    Column source_strings,
+    string_character_types types,
+    string_character_types verify_types
+):
+    """
+    Filter specific character types from a column of strings.
```
Comment (wence-) [inline python/pylibcudf/pylibcudf/strings/char_types.pyx:22]: This docstring is wrong I think?

File: python/pylibcudf/pylibcudf/strings/char_types.pyx [hunk 2]
```diff
@@ -1,4 +1,92 @@
...
+    replacement : Scalar
+        The replacement character to use when removing characters
+    types_to_keep : StringCharacterTypes
+        Default `ALL_TYPES` means all characters of `types_to_remove`
+        will be filtered.
+
+    Returns
+    -------
+    Column
+        New column of boolean results for each string
```
Comment (wence-) [inline python/pylibcudf/pylibcudf/strings/char_types.pyx:75]: This is wrong, it returns a new strings column with the requested characters filtered out and replaced with replacement.

I think you've copied this from the C++ docstring, which, it turns out, is also wrong, could you update that one too please?

**Review by wence- — CHANGES_REQUESTED** [review]
Some small nits/suggestions. Code looks good but some docstrings are incorrect I think.

File: python/pylibcudf/pylibcudf/strings/char_types.pyx (hunk 1 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/strings/char_types.pyx:22]: In libcudf, its `"Returns a boolean column identifying strings entries in which all characters are of the type specified."` Maybe we can simplify the libcudf and pylibcudf doc to be "A boolean column indicating whether all characters in each string are of the specified type."?

**Comment by Matt711** [comment]
Sorry for the noise. I'll clean up this PR.

File: python/pylibcudf/pylibcudf/tests/test_string_char_types.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pyarrow.compute as pc
+import pylibcudf as plc
```
Comment (Matt711) [inline python/pylibcudf/pylibcudf/tests/test_string_char_types.py:5]:
```suggestion
from pylibcudf.strings import char_types
```

**Review by wence- — APPROVED** [review]
Thanks!

File: python/pylibcudf/pylibcudf/tests/test_string_char_types.py (hunk 3 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/tests/test_string_char_types.py:5]: We're using `plc.` in other places here so this change isn't helpful

**Review by Matt711 — APPROVED** [review]

File: cpp/include/cudf/strings/char_types/char_types.hpp [hunk 4]
```diff
@@ -105,7 +105,8 @@ std::unique_ptr<column> all_characters_of_type(
  *        `types_to_remove` will be filtered.
  * @param mr Device memory resource used to allocate the returned column's device memory
  * @param stream CUDA stream used for device memory operations and kernel launches
- * @return New column of boolean results for each string
+ * @return New strings column with the specified characters filtered out and replaced with specified
+ * replacement string.
```
Comment (ttnghia) [inline cpp/include/cudf/strings/char_types/char_types.hpp:109]:
```suggestion
 * @return New strings column with the specified character types filtered out and replaced with specified
 * replacement string.
```

**Review by ttnghia — APPROVED** [review]

**Pull request merged**